### PR TITLE
[scripts] Add eos-merge-mcmc

### LINKED ISSUE #28: Add eos-merge-mcmc script
Should implement the basic functionality of our internal (non-public) script.

### PR DESCRIPTION
**Pull request by fredRos**
First working version from uncluttering `merge.py` from the old eos scripts repo

fixes #28

### CODE REVIEW

**Comment by fredRos** [comment]
* I didn't feel it was necessary to use the `MCMCDataFile` structure. Let me me know if you insist that I do. It was easier w/o it.
* To create test MCMC files, I copied from the manual. Now I remembered again why for BAT we wanted an HTML manual: it's much easier to copy&paste than from multi-line PDF. For me, the prerun needed 16000 iterations. I don't see why you argue in the manual that it should only take 7500 iterations. Perhaps that's what it happened to take on your machine. Also I was surprised that the results were deterministic, I didn't see a fixing of seeds in my call to `eos-sample-mcmc`. Is that on purpose?

**Comment by dvandyk** [comment]
Can you document it in the manual, and modify the examples to make use of it?

**Comment by dvandyk** [comment]
> * I didn't feel it was necessary to use the `MCMCDataFile` structure. Let me me know if you insist that I do. It was easier w/o it.

Nope, I am perfectly happy with it as it is.
> * To create test MCMC files, I copied from the manual. Now I remembered again why for BAT we wanted an HTML manual: it's much easier to copy&paste than from multi-line PDF.

Yes, it's a nuisance that some PDF viewers inject random whitespaces in the copied text.
Or is that a fault of LaTeX?

>  For me, the prerun needed 16000 iterations. I don't see why you argue in the manual that it should only take 7500 iterations. Perhaps that's what it happened to take on your machine.

It is rather meant as an example that you require only a small number of MCMC events to initialize the PMC run.

> Also I was surprised that the results were deterministic, I didn't see a fixing of seeds in my call to `eos-sample-mcmc`. Is that on purpose?

Not really. We should investigate that.

**Comment by fredRos** [comment]
The documentation of `eos-merge-mcmc` is in now. I still have to 

* [ ] check seed and reproducibility
* [ ] update manual on 7500 iterations
* [ ] check manual output. I couldn't build it due to this error

```

! Class scrbook Error: undefined old font command `\rm'.

See the scrbook class documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.328   {\cal L}_{\rm
                      EFT} &
```

**Comment by dvandyk** [comment]
> The documentation of eos-merge-mcmc is in now.

Thanks!

> I still have to
> - [ ]  check seed and reproducibility

Ok. That should be easy. The clients do only choose a random seed if ```--seed time``` is given.

> - [ ]  update manual on 7500 iterations

Yes please.

> - [ ]  check manual output. I couldn't build it due to this error

Oh, I see. We had that warning for quite some time. Seems that Komascript followed through with
removing ```\rm```. I think this is on @cBobeth though, since it is confined to ```manual/physics/eft.tex```:

```
./physics/eft.tex:328:  {\cal L}_{\rm EFT} &
./physics/eft.tex:329:  = {\cal L}_{{\rm QCD} \times {\rm QED}} (q;\, \ell) - \frac{4 G_F}{\sqrt{2}} V_{cb}^\text{eff} \left[
```

File: manual/doc/usage.tex [hunk 1]
```diff
@@ -352,10 +352,34 @@ \section{Producing Random Parameter Samples}
...
+
+To run multiple Markov chains in parallel, one may use a compute cluster and submit
+independent jobs, each of which creates its own output file with one or more
+chains inside. To initialize PMC, it is necessary to have all Markov chains in a
+single HDF5 file. This is where \client{eos-merge-mcmc} comes into play. It
+allows one to merge chains from files into a single HDF5 file and optionally to
+apply a cut such that chains that got stuck in irrelevant local modes can be
+filtered out.
+
+\clien{eos-merge-mcmc} accepts the following command-line arguments:
```
Comment (dvandyk) [inline manual/doc/usage.tex]: ```\client```

File: manual/doc/usage.tex [hunk 2]
```diff
@@ -352,10 +352,34 @@ \section{Producing Random Parameter Samples}
...
+\begin{itemize}
+    \item[] \cli{Any number of input files created by \cli{eos-sample-mcmc}}\\[-3\medskipamount]
```
Comment (dvandyk) [inline manual/doc/usage.tex]: Why is this wrapped in ```\cli{...}```?

**Comment by fredRos** [comment]
I'm done. I can build the manual on ubuntu 16.04. @cBobeth is still invited to remove `\rm` so it can be built on ubuntu in the future with more recent version of koma

**Comment by fredRos** [comment]
It seems the jobs didn't start on travis... the last build for this PR is from last night https://travis-ci.org/eos/eos/builds/379933236

**Pull request merged**